var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -102,15 +135,37 @@ def die_without_fortran(self):
                 'Mpich requires both C and Fortran compilers!'
             )
 
+    def with_or_without_tcp(self, activated):
```

Comment:
i don't see where you use those `with_or_without_xyz` functions?

Reply:
AFAICT, they get indirectly called by `config_args.extend(self.with_or_without('netmod'))`?

Reply:
my knowledge of python is limited, maybe @alalazo can comment. I am a bit confused with all those `with_or_without_` functions but probably that's a correct way of implementing this.

Reply:
This looks good: for a multivalued variant this would normally produce options like `--with-tcp`, `--with-mxm`... so these overrides are required. The `with_or_without('netmod')` call will call these functions.